components/AddTask.js

```diff
@@ -1,17 +1,54 @@
-export default function AddTask() {
+import { useState } from 'react'
+import axios from '../utils/axios'
+import { useAuth } from '../context/auth'
+import notif from '../components/Notif';
+
+export default function AddTask({tasks,setTasks}) {
+
+  const { token } = useAuth()
+  const [ text, setText] = useState('')
   const addTask = () => {
-    /**
-     * @todo Complete this function.
-     * @todo 1. Send the request to add the task to the backend server.
-     * @todo 2. Add the task in the dom.
-     */
-  }
+
+    if(!text){
+      notif('Error','Field is empty','danger')
+      return
+    }
+    
+    const dataForApiRequest = {
+      title: text
+    }
+
+    axios.post('todo/create/',
+          dataForApiRequest,{
+          headers: {
+            Authorization: 'Token ' + token,
+          },
+        })
+        .then((response) => {
+          notif('Success','Task added successfully','success')
+          axios.get('todo/', {
```

Comment:
Instead of again writing logic for getting tasks, you can make a function and use it wherever required